Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -162,7 +162,17 @@ template <class Type> class TableColumn : public API::Column {
         typename boost::mpl::if_c<boost::is_convertible<double, Type>::value,
                                   Type, InconvertibleToDoubleType>::type
             DoubleType;
-    return boost::numeric_cast<double, DoubleType>(m_data[i]);
+    if (typeid(m_data[i]) == typeid(std::string)) {
+      std::string trimmedString = boost::lexical_cast<std::string>(
+          m_data[i]); // so not to alter the original data
+
+      if (trimmedString.back() == ':') {
```

Comment:
I don't understand the need for this?

Our cases are essentially whether we can directly cast from `Type` to a `double` or if we have a string type that we need to use `lexical_cast` on. If Type is `string` and that string cannot be interpreted as a number then we should just throw shouldn't we?